src/transformation/utils/safe-names.ts

```diff
@@ -70,32 +70,42 @@ function checkName(context: TransformationContext, name: string, node: ts.Node):
 export function hasUnsafeSymbolName(
     context: TransformationContext,
     symbol: ts.Symbol,
-    tsOriginal: ts.Identifier
+    tsOriginal: ts.Identifier,
+    luaName = symbol.name
```

Comment:
I really dislike the signature of this function changing for an annotation, same for hasUnsafeIdentifierName, since you have the symbol and identifier nodes, I think it would be nicer to so `getLuaName` inside, or even just avoiding calling these functions if a luaName was specified. I would prefer to keep the impact of this change limited as much as possible to just annotations.ts, with minor adjustments to call.ts and identifier.ts.

Reply:
Sorry, was unsure exactly what the guarantees were for these arguments. Unfamiliar codebase syndrome. I'll apply your suggestion ASAP